modules/sonobiBidAdapter.js

```diff
@@ -117,9 +116,21 @@ export const spec = {
       payload.schain = JSON.stringify(validBidRequests[0].schain)
     }
     if (deepAccess(validBidRequests[0], 'userId') && Object.keys(validBidRequests[0].userId).length > 0) {
-      payload.userid = JSON.stringify(validBidRequests[0].userId);
+
+      const userIds = validBidRequests[0].userId;
+      if(userIds.id5id) {
+        userIds.id5id = deepAccess(userIds, 'id5id.uid');
```

Comment:
In general, this is not a good practice, you're modifying the source object here, but in this case, it should be safe as I tested it (and checked the core code) and it appears you're getting an adapter-specific copy of validBidRequests.

Reply:
Good catch. I copied the userId object so it doesnt get altered when updating id5id here.

Reply:
are you sure that this way you're actually cloning the object properly? Check this example:
![image](https://user-images.githubusercontent.com/12955750/110474190-ac32ba00-80df-11eb-88c8-dcd0e5f6a191.png)
There's utils.deepClone() inside pbjs that you can utilize for cloning.